### PR TITLE
Update GitHub org from pydata to pandas-dev

### PR DESCRIPTION
**Pull request by dhimmel**
It looks like the repo moved owners from `pydata` to `pandas-dev`. This pull request updates the repository to refer to `pandas-dev/pandas` rather than `pydata/pandas`.

- [x] passes ``git diff upstream/master | flake8 --diff``
- [ ] tests added / passed
- [ ] whatsnew entry

### CODE REVIEW

**Comment by dhimmel** [comment]
I didn't replace the following two URLs:

+ https://gitter.im/pydata/pandas
+ http://scatterci.github.io/pydata/pandas

I did replace what's new URLs -- not sure if "what's new" files should be immutable after release?

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -143,7 +143,7 @@ both "column wise min/max and global min/max coloring."
 API
 -----
 
-`pandas-datareader <https://github.com/pydata/pandas-datareader>`__
+`pandas-datareader <https://github.com/pandas-dev/pandas-datareader>`__
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst]: this one is still in pydata

Comment (dhimmel) [inline doc/source/ecosystem.rst]: Addressed in 2368761a396b2fa00a6201c46e0ca8702eab9d60

File: pandas/io/wb.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 raise ImportError(
     "The pandas.io.wb module is moved to a separate package "
     "(pandas-datareader). After installing the pandas-datareader package "
-    "(https://github.com/pandas-dev/pandas-datareader), you can change "
+    "(https://github.com/pydata/pandas-datareader), you can change "
```
Comment (jreback) [inline pandas/io/wb.py]: remove this

File: pandas/io/data.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 raise ImportError(
     "The pandas.io.data module is moved to a separate package "
     "(pandas-datareader). After installing the pandas-datareader package "
```
Comment (jreback) [inline pandas/io/data.py]: remove this

File: doc/source/remote_data.rst [hunk 4]
```diff
@@ -13,7 +13,7 @@ DataReader
 
 The sub-package ``pandas.io.data`` is removed in favor of a separately
 installable `pandas-datareader package
-<https://github.com/pandas-dev/pandas-datareader>`_. This will allow the data
+<https://github.com/pydata/pandas-datareader>`_. This will allow the data
```
Comment (jreback) [inline doc/source/remote_data.rst]: remote this

File: doc/source/contributing.rst [hunk 5]
```diff
@@ -27,7 +27,7 @@ about it!
 
 Feel free to ask questions on the `mailing list
 <https://groups.google.com/forum/?fromgroups#!forum/pydata>`_ or on `Gitter
```
Comment (jreback) [inline doc/source/contributing.rst]: remove this

**Comment by jreback** [comment]
@dhimmel there are only very very few changes remaining, you are adding files that are already deleted

**Comment by dhimmel** [comment]
> you are adding files that are already deleted

@jreback sorry I'm confused. I didn't think I was adding any new files? Where have they been deleted?

Do you also want me to doc updates, like removing the Google Group Mailing List link? That seems unrelated to this PR, but I'm happy to make those changes here if desired.

**Comment by jreback** [comment]
looks at the changes, some were changed from pandas-dev TO pydata (and not the reverse)

**Comment by jorisvandenbossche** [comment]
@jreback The ones that are changed back to pydata, are the pandas-datareader ones, so it is correct to change this back

**Comment by dhimmel** [comment]
@jreback see https://github.com/pandas-dev/pandas/pull/14575#discussion_r86357051: `pandas-datareader` is still hosted at pydata. 

> looks at the changes, some were changed from pandas-dev TO pydata (and not the reverse)

Check the diff for the entire PR.

**Comment by jorisvandenbossche** [comment]
Thanks @dhimmel

**Pull request merged**